libraries/utilities/include/graphene/utilities/file_util.hpp

```diff
@@ -23,12 +21,18 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#pragma once
 
-#include <fstream>
 #include <string>
 
 namespace graphene { namespace utilities {
 
+   /**
+    * @brief Reads the file at the given path and returns the full contents in a string. Note that the result
+    *        will contain non-printable characters if the file does.
```

Comment:
When I saw "non-printable characters", '\0' didn't come to my mind. I don't know whether it's only me though. Perhaps better if we explicitly mention '\0'? Callers need to be careful when using this function, since functions like `c_str()` are heavily used in the code base, a `strcpy()` or `strlen()` call on the result of `c_str()` may return unexpected results.